Modules/_datetimemodule.c

```diff
@@ -32,6 +32,8 @@
 #define PyTZInfo_Check(op) PyObject_TypeCheck(op, &PyDateTime_TZInfoType)
 #define PyTZInfo_CheckExact(op) (Py_TYPE(op) == &PyDateTime_TZInfoType)
 
+#define PyTimezone_Check(op) PyObject_TypeCheck(op, &PyDateTime_TimeZoneType)
```

Comment:
I believe that not having a `PyTimezone_Check` was a deliberate omission by @abalkin and that `timezone` is not intended to be subclassed.

I guess this isn't public anyway, but considering that it's not, I think there's no need to define both `PyTimezone_Check` and `PyTimezone_CheckExact`, just define whichever one you need. I'm thinking `PyTimezone_CheckExact` is most in keeping with the "`timezone` shouldn't be subclassed" philosophy (we can always add subclass support later, after all, while taking it away would be a PITA).

Reply:
`PyTZInfo_Check` was used elsewhere and I prefer that this PR is not touching those functions. I can inline the macro if you prefer